examples/launch_flatten.cpp

```diff
@@ -97,13 +102,12 @@ int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
   RAJA::View<int, RAJA::Layout<2>> d_A_2DView(d_A_ptr, N, N);
   RAJA::View<int, RAJA::Layout<1>> d_A_1DView(d_A_ptr, NN);
 
-
-  RAJA::expt::launch<device_launch>
-    (grid, [=] RAJA_HOST_DEVICE (RAJA::expt::LaunchContext ctx)
+  RAJA::launch<device_launch>
+    (dynamic_shared_mem, grid, [=] RAJA_HOST_DEVICE (RAJA::LaunchContext ctx)
```

Comment:
Does launch now always require shmem as an arg? It would be nice if it was optional.
Maybe if it used a special type it would be easier to make optional?
```suggestion
    (RAJA::DynamicSharedMemory(dynamic_shared_mem), grid, [=] RAJA_HOST_DEVICE (RAJA::LaunchContext ctx)
```

Reply:
yes, I like both suggestions.

Reply:
How can we make it optional with the special type? I'm thinking it would be easy to make optional if it was the last argument of launch, but not sure if it was not.

Reply:
@artv3 would it help or make sense to pass the shmem to the LaunchContext ctor, optionally?

Reply:
Or maybe have a single general front-end that takes a parameter pack when we could have a function that would find a specific type from the parameter pack or return a default if not found.

Reply:
That would work too.  To me, it seems that LaunchContext should take on more responsibilities.

Reply:
Actually passing it into the launch context won't work since in line 106 that is simply a function signature, what we could do is pass it into "Grid" as an optional parameter. What do you guys think?